### ISSUE TITLE
Support for DateTime when writing XLSX

### ISSUE THREAD

**Issue by stof** [box/spout#662]
The XLSX writer (and maybe also ODS, I haven't check) is missing support for date cells.

**Comment by mcorteel-harel** [comment]
There is an inconsistency in the code I think.
In `Box\Spout\Writer\XLSX\Manager\WorksheetManager::getCellXml` and `Box\Spout\Writer\ODS\Manager\WorksheetManager::getCellXml`, the case `elseif ($cell->isDate())` is not implemented, though `Box\Spout\Common\Entity\Cell` supports it.
Also, when exporting to CSV, `Cell::__toString()` tries to cast a `DateTime` to a string, which is not possible and produces an exception.

**Comment by schrieveslaach** [comment]
#735 adds this to ODT.

**Comment by ChronicLogic** [comment]
PR #751 adds XLSX.

**Comment by adrilo** [comment]
> There is an inconsistency in the code I think.
> In `Box\Spout\Writer\XLSX\Manager\WorksheetManager::getCellXml` and `Box\Spout\Writer\ODS\Manager\WorksheetManager::getCellXml`, the case `elseif ($cell->isDate())` is not implemented, though `Box\Spout\Common\Entity\Cell` supports it.
> Also, when exporting to CSV, `Cell::__toString()` tries to cast a `DateTime` to a string, which is not possible and produces an exception.

This is true. `Cell` could support `Date` objects but full support is not ready.

**Comment by mu1f407** [comment]
Hi, I recently faced this issue of missing support for dates for XLSX in my project. I saw that there were some PRs for this. I think the nicest one was #816. For me this solution is great and it's according to specification. The fact that some clients (Google Docs was mentioned) don't support it doesn't bother me. This shouln't be the reason for not merging this functionality. It works in Excel, is according to spec and it doesn't break anything.

@adrilo Would it be possible to reconsider merge of this PR? Thanks.

**Comment by Slamdunk** [comment]
Solved in https://github.com/openspout/openspout/pull/13

**Comment by mcorteel-harel** [comment]
@Slamdunk Thanks! Any idea when this will be released?

**Comment by Slamdunk** [comment]
Released now in [`v3.6.0`](https://github.com/openspout/openspout/releases/tag/v3.6.0)

Be aware that you need to specify the number format to have the output you need.

**Comment by mcorteel-harel** [comment]
@Slamdunk Could you give an example? Would that work (seems weird)?

```php
$style = (new StyleBuilder())
    ->setFormat('0.000');
$cell = WriterEntityFactory::createCell(new \DateTime(), $style);
```

**Comment by Slamdunk** [comment]
```php
$style = (new StyleBuilder())
    ->setFormat('yyyy-mm-dd');
$cell = WriterEntityFactory::createCell(new \DateTime(), $style);
```

**Comment by mcorteel-harel** [comment]
:+1: Thanks a lot!
